messaging/src/main/java/org/axonframework/eventhandling/TrackingEventProcessor.java

```diff
@@ -162,6 +160,71 @@ public void start() {
         }
     }
 
+    /**
+     * Instruct the processor to split the segment with given {@code segmentId} into two segments, allowing an
+     * additional thread to start processing events concurrently.
+     * <p>
+     * To be able to split segments, the {@link TokenStore} configured with this processor must use explicitly
+     * initialized tokens. See {@link TokenStore#requiresExplicitInitialization()}. Also, the given {@code segmentId}
+     * must be currently processed by a thread owned by this processor instance.
+     *
+     * @param segmentId The identifier of the segment to split
+     * @return a CompletableFuture providing the result of the split operation
+     */
+    public CompletableFuture<Boolean> splitSegment(int segmentId) {
+        if (!this.activeSegments.containsKey(segmentId)) {
+            return CompletableFuture.completedFuture(false);
+        }
+
+        CompletableFuture<Boolean> result = new CompletableFuture<>();
+        if (!tokenStore.requiresExplicitInitialization()) {
+            result.completeExceptionally(new UnsupportedOperationException("TokenStore must use explicit initialization to be able to safely split tokens"));
```

Comment:
Shouldn't `result` be returned immediately after the `completeExceptionally` call, just as with the `mergeSegment(int)` function?